packages/apollo-gateway/src/executeQueryPlan.ts

```diff
@@ -359,19 +356,16 @@ function mapFetchNodeToVariableDefinitions(
     return [];
   }
 
-  const variableMap = variableUsage.reduce((map, { node, type }) => {
-    const key = `${node.name.value}_${type.toString()}`;
-
-    if (!map.has(key)) {
-      map.set(key, {
-        kind: Kind.VARIABLE_DEFINITION,
-        variable: node,
-        type: astFromType(type),
-      });
-    }
+  const variableMap = Object.entries(variableUsage).reduce(
+    (map, [name, node]) => {
+      if (!map.has(name)) {
+        map.set(name, node);
+      }
 
-    return map;
-  }, new Map<string, VariableDefinitionNode>());
+      return map;
+    },
+    new Map<string, VariableDefinitionNode>(),
```

Comment:
Hmmm, I think we may not actually need a map here because `getVariableUsages()` only returns one entry per variable name.

Reply:
Good catch!